src/emulator/util/src/util.cpp

```diff
@@ -71,14 +72,21 @@ void init() {
         std::cerr << "spdlog error: " << msg << std::endl;
     });
 
-    spdlog::set_pattern("%^[%H:%M:%S.%e] |%L| [%!]: %v%$");
+    spdlog::set_pattern(LOG_PATTERN);
     spdlog::flush_on(spdlog::level::debug);
 }
 
 void set_level(spdlog::level::level_enum log_level) {
     spdlog::set_level(log_level);
 }
 
+void add_sink(std::wstring log_path) {
+    sinks.push_back(std::make_shared<spdlog::sinks::basic_file_sink_mt>(log_path, true));
```

Comment:
macOS build fails because `basic_file_sink_mt` accepts `std::string` and not `std::wstring`
do an `#ifdef _WIN32` with the above line there, and `utf16_to_utf8(log_path)` instead of `log_path` for the `#else`
need to find a neater way to deal with this stuff...